util/dnsname/dnsname.go

```diff
@@ -65,6 +60,17 @@ func ToFQDN(s string) (FQDN, error) {
 	return FQDN(raw), nil
 }
 
+func ToFQDN(s string) (FQDN, error) {
+	if len(s) == 0 || s == "." {
+		return FQDN("."), nil
+	}
+
+	if s[0] == '.' {
+		s = s[1:]
+	}
+	return ToFQDNSuffix(s)
```

Comment:
I see this is taking on the role of a general constructor. I think the API would be more clear if we had:

NewFQDN  // creates an FQDN without validating leading '.' presence, it accepts with or without. It adds a trailing '.' if one is not present.
ToFQDN   // ensures that there is no leading '.', removing it if found, otherwise delegates to NewFQDN.
ToFQDNSuffix // ensures that there is a leading '.', adding it if not found, otherwise delegates to NewFQDN.

The only problem currently is that ToFQDN has opinionated behavior (stripping '.'), but ToFQDNSuffix has unopinionated behavior (ignoring leading '.' presence), but if you were to just make assumptions based on the names, it would seem like the more specific name `ToFQDNSuffix` would have a more specific behavior.